ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor.hs

```diff
@@ -558,25 +579,103 @@ peerSelectionGovernorLoop tracer
       -- get the current time after the governor returned from the blocking
       -- 'evalGuardedDecisions' call.
       now <- getMonotonicTime
-      let Decision { decisionTrace, decisionJobs, decisionState } =
+
+      let Decision { decisionTrace, decisionJobs, decisionState = st'' } =
             timedDecision now
 
       mbCounters <- atomically $ do
+        -- Update outbound connections state
+        let peerSelectionView = peerSelectionStateToView st''
+            PeerSelectionView {
+                viewEstablishedPeers,
+                viewEstablishedBootstrapPeers,
+                viewActiveBootstrapPeers
+              }
+              =
+              fst <$> peerSelectionView
+            PeerSelectionCounters {
+                numberOfActiveBigLedgerPeers = _a
+              } = snd <$> peerSelectionView
+            PeerSelectionTargets {
+                targetNumberOfActiveBigLedgerPeers = _b
+              } = targets st''
+
+            trustableLocalRootSet = LocalRootPeers.trustableKeysSet (localRootPeers st'')
+
+        isPromiscuous <- readPromiscuousMode interfaces
+                                             (bootstrapPeersFlag st'')
+                                             (peerSelectionStateToView st'')
+
+        let !outboundConnectionsState =
+              case (isPromiscuous, bootstrapPeersFlag st'') of
+                (False, _) -> TrustedStateWithExternalPeers
```

Comment:
I think this case must consider `trustableLocalRootSet` . The node may not be promiscuous but it may still be talking to untrusted localroots.

Reply:
Done